server/src/main/java/org/elasticsearch/common/blobstore/BlobContainer.java

```diff
@@ -37,39 +37,55 @@ public interface BlobContainer {
     /**
      * Tests whether a blob with the given blob name exists in the container.
      *
-     * @param   blobName
-     *          The name of the blob whose existence is to be determined.
-     * @return  {@code true} if a blob exists in the {@link BlobContainer} with the given name, and {@code false} otherwise.
+     * @param purpose The purpose of the operation, useful for stats collection.
```

Comment:
I think there are several potential uses of this, stats collection, storage classes, priority. Perhaps we can move the examples to `OperationPurpose` javadoc and leave the example out here (and in other methods in this class)?

```suggestion
     * @param purpose The purpose of the operation
```

Reply:
Makes sense. I removed it from individual method and added it (with a bit more explanation) to the Enum class. Btw, the other usages sound pretty exciting!